### PR TITLE
Add Min duration based conditional sampler.

### PR DESCRIPTION
**Pull request by akodali18**

### CODE REVIEW

File: src/test/java/com/wavefront/sdk/entities/tracing/sampling/MinConditionalSamplerTest.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  private static final long SPAN_MIN_DURATION = 10;
+  MinConditionalSampler minConditionalSampler = new MinConditionalSampler(new ArrayList<>(),
+      SPAN_MIN_DURATION);
+
+  /**
+   * Test valid span i.e. span more than minimum duration gets accepted.
+   */
+  @Test
+  public void testValidSpan() {
+    assertTrue(minConditionalSampler.sample(SPAN_OP_NAME,1231, 13),
```
Comment (MikeMcMahon) [inline src/test/java/com/wavefront/sdk/entities/tracing/sampling/MinConditionalSamplerTest.java:25]: nit. formatting

File: src/test/java/com/wavefront/sdk/entities/tracing/sampling/MinConditionalSamplerTest.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        "Span duration is greater than minimum threshold and expected to be sampled.");
+
+  }
+
+  /**
+   * Test Invalid span i.e. span less than minimum duration gets discarded.
+   */
+  @Test
+  public void testInvalidSpan() {
+    assertFalse(minConditionalSampler.sample(SPAN_OP_NAME,1231, 8),
```
Comment (MikeMcMahon) [inline src/test/java/com/wavefront/sdk/entities/tracing/sampling/MinConditionalSamplerTest.java:35]: nit. formatting

File: src/test/java/com/wavefront/sdk/entities/tracing/sampling/MinConditionalSamplerTest.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+import java.util.ArrayList;
+
+/**
+ * @author Anil Kodali (akodali@vmware.com).
+ */
+public class MinConditionalSamplerTest {
+
+  private static final String SPAN_OP_NAME = "spanOperationName";
+  private static final long SPAN_MIN_DURATION = 10;
```
Comment (sushantdewan123) [inline src/test/java/com/wavefront/sdk/entities/tracing/sampling/MinConditionalSamplerTest.java:16]: SPAN_MIN_DURATION_MS
assuming this is 10 milliseconds?

File: src/main/java/com/wavefront/sdk/entities/tracing/sampling/MinConditionalSampler.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+/**
+ * This an extension of {@link CompositeSampler} that runs a pre-check to determine if a span is
+ * eligible for sampling. This implementation needs span to meet the minimum duration threshold
+ * to be considered for sampling, otherwise span is discarded.
+ *
+ * @author Anil Kodali (akodali@vmware.com)
+ */
+public class MinConditionalSampler extends CompositeSampler{
+
+  private final long minimumDuration;
```
Comment (sushantdewan123) [inline src/main/java/com/wavefront/sdk/entities/tracing/sampling/MinConditionalSampler.java:14]: duration should always have a unit
rename this to minimumDurationMillis

Comment (sushantdewan123) [reply src/main/java/com/wavefront/sdk/entities/tracing/sampling/MinConditionalSampler.java:14]: In hindsight, we should have renamed this variable to durationMillis
https://github.com/wavefrontHQ/wavefront-sdk-java/blob/master/src/main/java/com/wavefront/sdk/entities/tracing/sampling/DurationSampler.java#L10

**Pull request merged**